Hints.py

```diff
@@ -1116,7 +1117,12 @@ def buildWorldGossipHints(spoiler, world, checkedLocations=None):
             except IndexError:
                 raise Exception('Not enough valid hints to fill gossip stone locations.')
 
-        hint = hint_func[hint_type](spoiler, world, checkedLocations)
+        allCheckedLocations = set.union(checkedLocations, checkedAlwaysLocations)
+        if hint_type == 'barren':
+            hint = hint_func[hint_type](spoiler, world, checkedLocations)
+        else:
+            hint = hint_func[hint_type](spoiler, world, allCheckedLocations)
+        checkedLocations.update(allCheckedLocations.difference(set.union(checkedLocations, checkedAlwaysLocations)))
```

Comment:
This looks like it can be replaced with
```suggestion
        checkedLocations.update(allCheckedLocations - checkedAlwaysLocations)
```
since `update` is just `|=`, there's no need to add `checkedLocations` to the things to be subtracted out.
And it only really needs to occur in the else case if `hint_func` modified `allCheckedLocations`, otherwise this seems redundant.

Reply:
done